dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2130,161 @@ def _update_chi2(self):
+
+    def _calc_null_replication_metrics(self, clean_samples, prev_counts):
+        """
+        Calculate metrics needed for replicating null values in capitalone/synthetic-data.
+
+        Metrics include class priors and class specific means of each column.
+
+        Required for running LDA based binary classifier where predicted class label indicates whether
+        a value of a column should be NaN (1) or not (0).
+
+        :param clean_samples: input cleaned dataset
+        :type clean_samples: dict
+        :param prev_counts: null_count and sample_size of each column before update
+        :type prev_counts: dict
+        """
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors="coerce")
+
+        for col_id in range(len(self._profile)):
+            profile = self._profile[col_id]
+            null_count = getattr(profile, "null_count")
+            if null_count == 0:
+                # No missing values to replicate
+                continue
+
+            # Calculate class priors
+            # i.e probability that a value in target col is or is not NaN
+            sample_size = getattr(profile, "sample_size")
+            true_count = sample_size - null_count
+
+            # null_count and sample_size get updated on profile_update
+            # Therefore the priors are calculated from the updated values
+            prior_null = null_count / sample_size
+            prior_not_null = true_count / sample_size
+
+            # Partition data based on whether target column value is null or not
+            col_is_null = data.iloc[:, col_id].isnull()
+            partition_null = data[col_is_null]
+            partition_not_null = data[~col_is_null]
+
+            # Drop the target column from partitions
+            partition_null = partition_null.drop(columns=col_id)
+            partition_not_null = partition_not_null.drop(columns=col_id)
+
+            # Calculate mean vectors of each partition
+            # i.e mean vectors of rows where target col is or is not null
+            mean_not_null, mean_null = (
+                partition_not_null.mean().tolist(),
+                partition_null.mean().tolist(),
+            )
+
+            # Combine with old means
+            if col_id in self._null_replication_metrics:
+                prev_null_count, prev_sample_size = prev_counts[col_id]
+                prev_true_count = prev_sample_size - prev_null_count
+
+                # null_count, sample_size and true_count of new data chunk
+                added_null_count, added_sample_size = (
+                    null_count - prev_null_count,
+                    sample_size - prev_sample_size,
+                )
+                added_true_count = added_sample_size - added_null_count
+
+                prev_class_means = self._null_replication_metrics[col_id]["class_mean"]
+                prev_mean_not_null, prev_mean_null = prev_class_means
+
+                mean_not_null = utils.combine_means(
```

Comment:
What if we store sums and counts as oppose to re'calcing? Would that be better? Similar to how we store in numerical stats?